### PR TITLE
Added Counter based OTP (HOTP)

### PR DESCRIPTION
**Pull request by ResultsMayVary**
For my project I required a counter based solution. I've expanded this to support the counter_based flag to allow HOTP. It's been tested in my environment and is working as expected. Readme has been updated with usage instructions as well as the version number increase to v1.2.0.

Let me know if this is all good and if you want me to perform any other changes before pulling. Thanks!

### CODE REVIEW

**Comment by robertomiranda** [comment]
@ResultsMayVary thanks I'll check it during the weekend, in the mean time could you apply the changes suggested by @houndci?

**Comment by ResultsMayVary** [comment]
Of course. I was just reviewing them. Do you have a recommendation for the comments regarding lines being too long when it's coded the same as other lines?

**Comment by robertomiranda** [comment]
sure, follow this ruby style guide https://github.com/bbatsov/ruby-style-guide#no-double-indent

**Comment by robertomiranda** [comment]
btw, we should add that ruby style to the contributors guideline :grin:

File: lib/active_model/one_time_password.rb [hunk 1]
```diff
@@ -5,20 +5,25 @@ module OneTimePassword
...
         self.otp_column_name = (options[:column_name] || "otp_secret_key").to_s
         self.otp_digits = options[:length] || 6
 
+        self.otp_counter_based = (options[:counter_based] || false)
+        self.otp_counter_column_name = (options[:counter_column_name] || "otp_counter").to_s
+
         include InstanceMethodsOnActivation
 
-        before_create { self.otp_regenerate_secret if !self.otp_column}
+        before_create {
```
Comment (robertomiranda) [inline lib/active_model/one_time_password.rb]: right here would be better use, something like:

```ruby
before do
end
```

**Comment by ResultsMayVary** [comment]
I've fixed as many of the issues as I can. At this point what do I do about the redundant self issues? Not worry about them? If I don't access otp_counter as self it returns as ```false:FalseClass```.

**Comment by robertomiranda** [comment]
@ResultsMayVary sure, it doesn't matter

**Comment by robertomiranda** [comment]
@ResultsMayVary all looks good for me, but I'd like to have another opinion  @guilleiguaran could you check it?

**Comment by robertomiranda** [comment]
@ResultsMayVary thanks :+1:

**Comment by ResultsMayVary** [comment]
@robertomiranda welcome! I did the changes for work so I always contribute back when I can.

**Pull request merged**